scipy/special/_logsumexp.py

```diff
@@ -189,37 +186,36 @@ def _elements_and_indices_with_max_real(a, axis=-1, xp=None):
         max_i = xp.max(i, axis=axis, keepdims=True)
         mask = i == max_i
         a = xp.where(mask, a, 0.)
-        max = xp.sum(a, axis=axis, dtype=a.dtype, keepdims=True)
+        max_ = xp.sum(a, axis=axis, dtype=a.dtype, keepdims=True)
     else:
-        max = xp.max(a, axis=axis, keepdims=True)
-        mask = a == max
+        max_ = xp.max(a, axis=axis, keepdims=True)
+        mask = a == max_
 
-    return xp.asarray(max), xp.asarray(mask)
```

Comment:
This was probably here for a reason at some point due to NumPy returning scalars, which can't be mutated via `__setitem__`. I imagine that the change to `at.set` or something else makes this unnecessary.

Reply:
That's my understanding too.